src/test/java/com/google/research/bleth/simulator/ObserverTest.java

```diff
@@ -0,0 +1,930 @@
+
+        assertThat(staticObserver.getLocation()).isEqualTo(zeroOnZeroCoordinate);
+    }
+
+    @Test
+    public void randomObserverSurroundedByFourDirectionsStayOnItsLocation() {
+        // -----
+        // | O |
+        // -----
+        Board board = new Board(1, 1);
+        Mockito.when(simulation.getBoard()).thenReturn(board);
+        Observer randomObserver = createRandomObserverOnLocation(zeroOnZeroCoordinate);
+
+        randomObserver.move();
+
+        assertThat(randomObserver.getLocation()).isEqualTo(zeroOnZeroCoordinate);
+    }
+
+    @Test
+    public void randomObserverSurroundedByThreeDirectionsMovesUp() {
+        // -----
+        // |   |
+        // -----
+        // | O |
+        // -----
+        Board board = new Board(2, 1);
+        Mockito.when(simulation.getBoard()).thenReturn(board);
+        Observer randomObserver = createRandomObserverOnLocation(oneOnZeroCoordinate);
+
+        randomObserver.move();
+
+        assertThat(randomObserver.getLocation()).isEqualTo(zeroOnZeroCoordinate);
+    }
+
+    @Test
+    public void randomObserverSurroundedByThreeDirectionsMovesDown() {
+        // -----
+        // | O |
+        // -----
+        // |   |
+        // -----
+        Board board = new Board(2, 1);
+        Mockito.when(simulation.getBoard()).thenReturn(board);
+        Observer randomObserver = createRandomObserverOnLocation(zeroOnZeroCoordinate);
+
+        randomObserver.move();
+
+        assertThat(randomObserver.getLocation()).isEqualTo(oneOnZeroCoordinate);
+    }
+
+    @Test
+    public void randomObserverSurroundedByThreeDirectionsMovesRight() {
+        // ----------
+        // | O |    |
+        // ----------
+        Board board = new Board(1, 2);
+        Mockito.when(simulation.getBoard()).thenReturn(board);
+        Observer randomObserver = createRandomObserverOnLocation(zeroOnZeroCoordinate);
+
+        randomObserver.move();
+
+        assertThat(randomObserver.getLocation()).isEqualTo(zeroOnOneCoordinate);
+    }
+
+    @Test
+    public void randomObserverSurroundedByThreeDirectionsMovesLeft() {
+        // ----------
+        // |   | O  |
+        // ----------
+        Board board = new Board(1, 2);
+        Mockito.when(simulation.getBoard()).thenReturn(board);
+        Observer randomObserver = createRandomObserverOnLocation(zeroOnOneCoordinate);
+
+        randomObserver.move();
+
+        assertThat(randomObserver.getLocation()).isEqualTo(zeroOnZeroCoordinate);
+    }
+
+    @Test
+    public void randomObserverMoveExactlyOneStep() {
+        Board board = new Board(3, 3);
+        Mockito.when(simulation.getBoard()).thenReturn(board);
+        Observer randomObserver = createRandomObserverOnLocation(oneOnOneCoordinate);
+
+        randomObserver.move();
+
+        assertThat(calculateDistance(randomObserver.getLocation(), oneOnOneCoordinate)).isEqualTo(1);
+    }
+
+    @Test
+    public void randomObserverSurroundedByThreeDirectionsMoveExactlyOneStepAside() {
```

Comment:
Do you really need both this test and the one with a `Board(1, 2)`? This test covers both behaviors - (a) observer will move to the right, and (b) it will move just a single step.